### PR TITLE
[SDK] Add VolumePathCannotBeEmpty validator

### PR DESCRIPTION
**Pull request by mbdcos**

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import com.mesosphere.sdk.specification.TaskSpec;
+import com.mesosphere.sdk.specification.VolumeSpec;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.*;
+
+/**
+ * Validates that volume path is not empty if new ServiceSpec has a volume
+ */
+public class VolumePathCannotBeEmpty implements ConfigValidator<ServiceSpec> {
```
Comment (mohitsoni) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java]: Can you please add some unit tests ?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @Override
+    public Collection<ConfigValidationError> validate(ServiceSpec nullableOldConfig, ServiceSpec newConfig) {
+        List<ConfigValidationError> errors = new ArrayList<>();
+
+        for (PodSpec podSpec : newConfig.getPods()) {
+            for (TaskSpec taskSpec : podSpec.getTasks()) {
+                Collection<VolumeSpec> volumeSpecs = taskSpec.getResourceSet().getVolumes();
+                if (volumeSpecs.isEmpty()) {
```
Comment (mohitsoni) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java]: Can volumeSpecs be `null` ?

Comment (mbdcos) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java]: get ResourceSet returns a Collection. If no VolumeSpec, it should be an empty list.

Comment (mohitsoni) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java]: Okay

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java (hunk 1 above)
Comment (mbdcos) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java]: OK

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java (hunk 2 above)
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java]: Why have this check at all?  If it's empty the for loop will be a noop anyway.

Comment (mbdcos) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java]: It was checking null before that is why left here!

**Comment by mbdcos** [comment]
```@NotEmpty``` does not detect blank string

**Comment by mbdcos** [comment]
```@NotBlank``` works
We should also have ```@Pattern```

**Pull request closed without merging**